### PR TITLE
Frida complog fix

### PR DESCRIPTION
**Pull request by WorksButNotTested**

### CODE REVIEW

File: frida_mode/src/cmplog/cmplog.c [hunk 1]
```diff
@@ -27,6 +33,39 @@ static gint cmplog_sort(gconstpointer a, gconstpointer b) {
...
+
+  if (unlink(fname) < 0) {
+
+    FATAL("Failed to unlink temp file: %s (%d), errno: %d", fname, fd, errno);
+
+  }
+
+  if (ftruncate(fd, 0) < 0) {
+
+    FATAL("Failed to unlink temp file: %s (%d), errno: %d", fname, fd, errno);
```
Comment (vanhauser-thc) [inline frida_mode/src/cmplog/cmplog.c]: that would be truncate not unlink :-)

File: frida_mode/src/cmplog/cmplog.c [hunk 2]
```diff
@@ -81,6 +130,27 @@ gboolean cmplog_is_readable(guint64 addr, size_t size) {
...
+  /*
+   * Our address map can change (e.g. stack growth), use write as a fallback to
+   * validate our address.
+   */
+  ssize_t written = syscall(__NR_write, fd_null, (void *)addr, size);
+
+  /*
+   * Seek back to the start, we don't want our temp file to continue growing!
+   */
+  if (ftruncate(fd_null, 0) < 0) {
```
Comment (vanhauser-thc) [inline frida_mode/src/cmplog/cmplog.c]: you want to truncate for every write?? rather do that once per fork() or persistent run

File: frida_mode/src/cmplog/cmplog.c (hunk 1 above)
Comment (WorksButNotTested) [reply frida_mode/src/cmplog/cmplog.c]: Fixed.

File: frida_mode/src/cmplog/cmplog.c (hunk 2 above)
Comment (WorksButNotTested) [reply frida_mode/src/cmplog/cmplog.c]: I went with tracking the file size and truncating when it reached a given limit. It kept the interface between complog and the rest of the code more minimal and should still have the desired effect.

**Comment by vanhauser-thc** [comment]
LGTM
a few ideas:
  * increase the allowed size of the temp file. 1MB should be fine IMHO
  * use likely()/unlikely() in your if() for a bit more performance (everywhere in code that runs during fuzzing)
  * compile with perf and perform an analysis what is called very often, then use attribute(hot) for the most used functions, also gives quite a speed gain.

**Pull request merged**